src/measure_theory/measurable_space.lean

```diff
@@ -121,6 +121,14 @@ begin
   exact ‹measurable_space α›.is_measurable_Union _ (is_measurable.bUnion_decode2 h)
 end
 
+lemma is_measurable.Union_fintype [fintype β] {f : β → set α} (h : ∀ b, is_measurable (f b)) :
+  is_measurable (⋃ b, f b) :=
+begin
+  have := encodable.trunc_encodable_of_fintype β,
+  induction this using trunc.rec_on_subsingleton,
```

Comment:
This looks like a nasty trick, although this is exactly how it is supposed to be used if I understand correctly. Is it possible to have in the encodable file a nonconstructive def giving encodable out of `[fintype]`, and use it here with a `local attribute [instance]` (so that the proof would just be `is_measurable.Union h`). Like
```lean
noncomputable def fintype.encodable [fintype β] : encodable β :=
(encodable.trunc_encodable_of_fintype β).out
```

and

```lean
local attribute [instance] fintype.encodable

lemma is_measurable.Union_fintype [fintype β] {f : β → set α} (h : ∀ b, is_measurable (f b)) :
  is_measurable (⋃ b, f b) :=
is_measurable.Union h
```

(Then the proof below simplifies similarly).

Reply:
What you suggest is more user-friendly indeed.